clab/dependency_manager/dependency_manager.go

```diff
@@ -2,101 +2,30 @@ package dependency_manager
 
 import (
 	"fmt"
+	"sort"
 	"strings"
-	"sync"
 
 	log "github.com/sirupsen/logrus"
+	"github.com/srl-labs/containerlab/types"
 )
 
 type DependencyManager interface {
 	// AddNode adds a node to the dependency manager.
 	AddNode(name string)
 	// AddDependency adds a dependency between depender and dependee.
 	// The depender will effectively wait for the dependee to finish.
-	AddDependency(dependee, depender string) error
+	AddDependency(depender string, dependerState types.WaitForPhase, dependee string, dependeeState types.WaitForPhase) error
 	// WaitForNodeDependencies is called by a node that is meant to be created.
 	// This call will bock until all the nodes that this node depends on are created.
-	WaitForNodeDependencies(nodeName string) error
+	Enter(nodeName string, state types.WaitForPhase) error
 	// SignalDone is called by a node that has finished all tasks for the provided State.
 	// internally the dependent nodes will be "notified" that an additional (if multiple exist) dependency is satisfied.
-	SignalDone(nodeName string, state NodeState)
+	SignalDone(nodeName string, state types.WaitForPhase)
+	GetDependerCount(nodeName string, state types.WaitForPhase) (uint8, error)
```

Comment:
why uint8? maybe uint is better to not think about potential constraints

Reply:
ok

Reply:
Done -> uint